cxplat/inc/winuser/cxplat_platform.h

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+#pragma once
+// Do not add anything else to this file.
+// Actual platform-specific definitions go in the file below.
+#include "cxplat_winuser.h"
```

Comment:
```suggestion
#ifdef _KERNEL_MODE
#error "Windows kernel platform is unsupported"
#elif _WIN32
#include "cxplat_winuser.h"
#else
#error "Unknown unsupported Platform"
#endif
```
Maybe make this a bit explicit right off the bat?

Reply:
I have a general distaste for ifdefs, they clutter the code and can easily leave dead code sections.

Reply:
I don't feel strongly about this, so may adopt this suggestion in the next PR

Reply:
We can discuss more in a follow up PR, but I don't want each consumer of cxplat to be required to have platforms specific logic to change the include path. They should include one path, and just `#include <cxplat_platform.h>` and be good to go.

Reply:
Filed #83 to track